widget/embedded/src/utils/swap.ts

```diff
@@ -753,25 +751,43 @@ export function shouldRetrySwap(pendingSwap: PendingSwap) {
   return pendingSwap.status === 'failed';
 }
 
-export function confirmSwapDisabled(
+export function isConfirmSwapDisabled(
   fetching: boolean,
   showCustomDestination: boolean,
   customDestination: string | null,
   quote: SelectedQuote | null,
   selectedWallets: { walletType: string; chain: string }[],
   lastStepToBlockchain?: BlockchainMeta
-) {
+): boolean {
+  if (!quote || !lastStepToBlockchain || fetching) {
+    return true;
+  }
```

Comment:
You used `lastStepToBlockchain` to only check if a custom destination is valid or not. So it doesn't make sense to have this condition here even if it works. Because user may not enter a custom address.